spec/remote_io_spec.rb

```diff
@@ -7,7 +7,9 @@
     rio = described_class.new('https://images.invalid/img.jpg')
 
     fake_resp = double(headers: {'Content-Range' => '10-109/2577'}, status: 206, body: 'This is the response')
-    expect_any_instance_of(Faraday::Connection).to receive(:get).with('https://images.invalid/img.jpg', nil, range: 'bytes=10-109').and_return(fake_resp)
+    faraday_conn = instance_double(Faraday::Connection, get: fake_resp)
```

Comment:
You could extract the double in a `let` and the `allow` in a `before`, to DRY things up a bit 😊

Reply:
I tried but I think I can't do it.

Faraday is loaded only after calling `FormatParser::RemoteIO.new` ([here](https://github.com/WeTransfer/format_parser/blob/master/lib/remote_io.rb#L28)):
```ruby
def initialize(uri)
  require 'faraday'
```

I think it was done this way to avoid loading the gem when it's not necessary, for example, if you parse only local files.

So, when I tried to reference the Faraday class in the `before` block, I got the error `uninitialized constant Faraday`.

I think that it's worth to require the gem before just to change the test.

Is it ok for you?

Reply:
Ah, right...

Looking at it better, the subject could also be DRYed out, and that would also require the gem. It is more work than this PR covers, so I'm also fine with moving on with how it is at the moment.